streaming/storage/subscription.go

```diff
@@ -67,3 +69,49 @@ func (sub *Subscription) Close() error {
 		return err
 	}
 }
+
```

Comment:
This function does some weird stuff, but I guess the code kinda has to look this way.